File: language/src/main/java/com/deepbeginnings/coiffure/FileTypeDetector.java
package com.deepbeginnings.coiffure;

import com.oracle.truffle.api.TruffleFile;

import java.nio.charset.Charset;

public final class FileTypeDetector implements TruffleFile.FileTypeDetector {
    @Override
    public String findMimeType(TruffleFile file) {
        String name = file.getName();
        return (name != null && (name.endsWith(".clj") || name.endsWith(".cljc")))
                ? Language.MIME_TYPE
                : null;
    }

    @Override
    public Charset findEncoding(TruffleFile file) {
        return null;
    }
}

File: src/main/kotlin/FileTypeDetector.kt
import com.oracle.truffle.api.TruffleFile
import java.nio.charset.Charset

class FileTypeDetector : TruffleFile.FileTypeDetector {
    override fun findMimeType(file: TruffleFile?): String? {
        val name = file?.name
        return if (name != null && (name.endsWith(".clj") || name.endsWith(".cljc"))) {
            Coiffure.MIME_TYPE
        } else {
            null
        }
    }

    override fun findEncoding(file: TruffleFile?): Charset? = null
}
